src/FakeItEasy/Creation/DefaultFakeCreatorFacade.cs

```diff
@@ -35,6 +35,14 @@ public T CreateFake<T>(Action<IFakeOptions<T>> optionsBuilder)
             return (T)this.fakeAndDummyManager.CreateFake(typeof(T), options => optionsBuilder((IFakeOptions<T>)options));
         }
 
+        public object CreateFake(Type typeOfFake, Action<IFakeOptions> optionsBuilder)
+        {
+            Guard.AgainstNull(typeOfFake, nameof(typeOfFake));
+            Guard.AgainstNull(optionsBuilder, nameof(optionsBuilder));

```

Comment:
Ah, I wasn't aware of this rule. I saw a `Guard.AgainstNull` in the generic `CreateFake<T>`, so I added it here as well.